1

```diff
@@ -11,8 +11,13 @@ msg "Welcome to Void!"
 
 # Start core services: one-time system tasks.
 detect_virt
-for f in /etc/runit/core-services/*.sh; do
-	[ -r $f ] && . $f
+for cs in /etc/runit/core-services/*.sh; do
+    if [ -r "$cs" ]; then
+        kmsg "start:core:$cs"
+        . "$cs"
+        kmsg "finish:core:$cs"
+    fi
 done
+kmsg "finish:core"
```

Comment:
want a start:core for completeness?

Reply:
my thinking:
- we aren't guaranteed to have `/dev/kmsg` until after `00-pseudofs` runs
- we can use the first core service start line to get a rough start for all core services

so there's not much need for `start:core`